core/trino-main/src/test/java/io/trino/sql/TestExpressionInterpreter.java

```diff
@@ -336,6 +336,17 @@ public void testNegative()
         assertOptimizedEquals("-(bound_decimal_long-12345678901234567890.123)", "CAST(0 AS decimal(20,10))");
     }
 
+    @Test
+    public void testArithmeticUnary()
+    {
+        assertOptimizedEquals("-rand()", "-rand()");
+        assertOptimizedEquals("-(0 / 0)", "-(0 / 0)");
+        assertOptimizedEquals("-(-(-(0 / 0)))", "-(-(-(0 / 0)))");
```

Comment:
For later, we should add an optimization that removes redundant negative unary expressions. E.g. `-(-(x))` => `x`

Reply:
Added.